core/domain/platform_parameter_domain.py

```diff
@@ -288,7 +287,9 @@ def validate(self):
 
         if self._type == 'server_mode':
             for _, mode in self._conditions:
-                if mode not in ALLOWED_SERVER_MODES:
+                try:
+                    getattr(SERVER_MODES, mode)
```

Comment:
Should this be referring to ALLOWED_SERVER_MODES, instead? Ditto below for FEATURE_STAGES.

Reply:
Yes, I missed that out. Thanks for pointing it out. I have changed it.

Reply:
done